src/popover.ts

```diff
@@ -9,6 +11,37 @@ export function isSupported() {
 const notSupportedMessage =
   'Not supported on element that does not have valid popover attribute';
 
+export const ORIGINAL_ATTACH_SHADOW_SYMBOL: unique symbol = Symbol();
+
+function patchAttachShadow(callback: (shadowRoot: ShadowRoot) => void) {
+  Element.prototype[ORIGINAL_ATTACH_SHADOW_SYMBOL] =
+    Element.prototype.attachShadow;
+  Element.prototype.attachShadow = function (init) {
+    const shadowRoot = this[ORIGINAL_ATTACH_SHADOW_SYMBOL](init);
+    callback(shadowRoot);
+    return shadowRoot;
+  };
+}
```

Comment:
You could use a variable to capture this, and avoid having to add extra fields to Element.prototype which in turn needs changes to TS:

```suggestion
export const originalAttachShadow = Element.prototype.attachShadow

function patchAttachShadow(callback: (shadowRoot: ShadowRoot) => void) {
  Element.prototype.attachShadow = function (init) {
    const shadowRoot = originalAttachShadow.call(this, init);
    callback(shadowRoot);
    return shadowRoot;
  };
}
```

Reply:
strangely this simple solution failed when I initially set this up. well, now it charmlessly works

Reply:
scoped the const within function to allow multi patching